src/helpers.rs

```diff
@@ -66,32 +66,39 @@ pub fn encrypt(data: &[u8]) -> String {
     }
 }
 
-pub fn create_jwt(user_id: String, finger: Option<String>) -> String {
+pub async fn create_jwt(user_id: String, finger: Option<String>, nonce: Option<String>) -> String {
     match EncodingKey::from_rsa_pem(dotenv::var("RSA_PRIVATE_KEY").expect("Missing env `RSA_PRIVATE_KEY`").as_bytes()) {

```

Comment:
The instruction does read from env every time it is called, which could result in undefined behaviors.

It might be better and simpler to parse the required environment variables when the application is launching and then sharing the value using function parameters.

In case you do require a lot of environment variables or you have a complex configuration, I can recommend using the [config](https://docs.rs/config) crate